### ISSUE TITLE
For discussion: log of unmatched questions / inputs

### ISSUE THREAD

**Issue by bmann** [superscriptjs/superscript#320]
I was talking to @silentrob today and thinking about how to generate dialogue in a friendly way for non-developers.

One thought was that if you could turn on logging in such a way that the output is the skeleton of unmatched inputs, you could make it easy to then edit the log and import it.

Various logs over time would generate a skeleton of the unmatched areas which you could fill in. This isn't very sophisticated or reactive, but could be an easy way to get started, and would mean actually use the bot interface itself (any of them, although I'm again thinking of the easy use case of chatting in Slack) to train the bot.

For instance:

```
Me: What is the capital of British Columbia?
Bot: <no idea response>
Log: query: What is the capital of British Columbia? response: <blank>
```

That way, I could actually use the bot to generate what I want to teach it. I'd then open up the log in a text editor, fill in the responses, and run parse to add it to the database.

**Comment by thinkgareth** [comment]
Love this!

**Comment by pringlized** [comment]
+1. Fantastic idea!

**Comment by dcsan** [comment]
logged with the current topic so that content stays relevant

**Comment by silentrob** [comment]
I agree this is a great idea, as a stop gap the reply object has a debug object which shows the topic matched and all the gambits it traversed in construct the reply.
